app/pages/Accounts/AccountDetailsPage/CredentialInformation/CredentialInformation.tsx

```diff
@@ -52,20 +42,21 @@ export default function CredentialInformation({ account, accountInfo }: Props) {
     }
 
     const credentials = Object.values(accountInfo.accountCredentials)
-        .map((o) => o.value.contents)
+        .map((o) => o.value)
         .map((cred) => {
-            let enrichedCred: CredentialOfAccount = {
-                ...cred,
+            const enrichedCred = {
+                ...cred.contents,
+                // The node returns the credId in the regId field for
```

Comment:
Since we now have a check on the type, the comment is no longer necessary to me. It is no longer a hack, as we actually check instead of just take whatever field was set.